tests/c/simple_interp_loop2.newcg.c

```diff
@@ -13,7 +13,6 @@
 //     --- Begin jit-pre-opt ---
 //     ..~
 //     guard %{{_}}, true
-//     %{{_}}: ptr = load %{{8}}
```

Comment:
Do we know why this has disappeared?

Reply:
`%8` is loaded at the begining of the entry block, seems like `mem2reg` removed redudant load here.

Reply:
I understand the observation, but we need to understand *why* this is correct.

Reply:
Based on my understanding of mem2reg, this seems legit.

The purpose of mem2reg is to replace allocations (and loads and stores from/to them) with SSA variables and PHI nodes (if necessary).

https://llvm.org/docs/Passes.html#mem2reg-promote-memory-to-register